app/models/friend_request.rb

```diff
@@ -2,4 +2,97 @@ class FriendRequest < ApplicationRecord
 
   belongs_to :sender, class_name: "User"
   belongs_to :receiver, class_name: "User"
+
+  def self.current_user_sent_friend_requests(user)
+    user.sent_friend_requests.map do |request|
+      {
+        id: request.id,
+        receiver_id: request.receiver.id,
+        full_name: request.receiver.full_name
+      }
+    end
+  end
+
+  def self.current_user_received_friend_requests(user)
+    user.received_friend_requests.map do |request|
+      {
+        request_id: request.id,
+        sender_id: request.sender.id,
+        full_name: request.sender.full_name
+      }
+    end
+  end
+
+  def self.create_request(sender, receiver_id)
+    receiver = User.find_by(id: receiver_id)
+    return { status: :not_found, message: "The user you are trying to send a request to does not exist" } if receiver.nil?
+
+    if sender.id == receiver_id
+      return { status: :unprocessable_entity, message: "You cannot send a friend request to yourself" }
+    end
+
+    if sender.friend_lists.exists?(friend_id: receiver_id) || receiver.friend_lists.exists?(friend_id: sender.id)
+      return { status: :unprocessable_entity, message: "You are already friends with this user" }
+    end
+
+    if sender.sent_friend_requests.exists?(receiver_id: receiver_id)
+      return { status: :unprocessable_entity, message: "Friend request already sent to this user" }
+    end
+
+    if receiver.sent_friend_requests.exists?(receiver_id: sender.id)
+      return { status: :unprocessable_entity, message: "You have already received a friend request from this user" }
+    end
+
+    friend_request = sender.sent_friend_requests.new(receiver: receiver)
+    if friend_request.save
+      { status: :ok, message: "Friend request sent successfully" }
+    else
+      { status: :unprocessable_entity, message: "Failed to send friend request", errors: friend_request.errors.full_messages }
+    end
+  end
```

Comment:
The method is too large. Can you reduce this method by removing unnecessary checking?

Reply:
Moved the checkings to private class method.